From 95a979edbb2d592b02752d55b595b558d4c764fd Mon Sep 17 00:00:00 2001
From: lenxyang <lenxyang@gmail.com>
Date: Fri, 7 Nov 2014 20:05:07 +0800
Subject: [PATCH]  build

---
 BUILD.gn | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index 5d1bad5..e6137b9 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -196,6 +196,8 @@ if (is_win) {
     ]
 
     include_dirs = [ "src/libGLESv2" ]
+    deffile = rebase_path("src/libGLESv2/libGLESv2.def", "$root_build_dir")
+    ldflags = ["/DEF:$deffile"]
 
     deps = [
       ":commit_id",
@@ -207,6 +209,7 @@ if (is_win) {
 
   shared_library("libEGL") {
     sources = rebase_path(egl_gypi.angle_libegl_sources, ".", "src")
+    sources += rebase_path(egl_gypi.angle_libegl_win32_sources, ".", "src")
 
     defines = [
       "GL_APICALL=",
@@ -226,9 +229,18 @@ if (is_win) {
         "ANGLE_GENERATE_SHADER_DEBUG_INFO"
       ]
     }
+    if (angle_enable_d3d9) {
+      defines += [ "ANGLE_ENABLE_D3D9" ]
+    }
+
+    if (angle_enable_d3d11) {
+      defines += [ "ANGLE_ENABLE_D3D11" ]
+    }
 
     include_dirs = [ "src/libGLESv2" ]
     libs = [ "d3d9.lib" ]
+    deffile = rebase_path("src/libEGL/libEGL.def", "$root_build_dir")
+    ldflags = ["/DEF:$deffile"]
 
     deps = [
       ":commit_id",
-- 
1.8.4.3

